API/Endpoints/ExerciseEndpoints.cs

```diff
@@ -71,6 +72,17 @@ public static WebApplication UseExerciseEndpoints(this WebApplication app)
                 }
                 return TypedResults.NoContent();
             }).RequireAuthorization(nameof(Roles.Instructor));
+        
+        exerciseV1.MapPost("/{exerciseId:int}/submit", async Task<IResult> ([FromBody] SubmitSolutionDto dto, int exerciseId, ISolutionRunnerService service) =>
```

Comment:
Could consider making another url for the endpoint. This could be confused with the POST "\" endpoint.